ios/Classes/RingerModeDetector.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+import AudioToolbox
+
+public class RingerModeDetector: NSObject {
+
+    public static let instance = RingerModeDetector()
+
+    var soundId : SystemSoundID = 0
+
+    private static var soundUrl: URL {
+        guard let soundUrl = Bundle(for: RingerModeDetector.self).url(forResource: "silence", withExtension: "aiff") else {
+            fatalError("File silence.aiff not found")
+        }
+        return soundUrl
+    }
+        
+    private var startTime : CFTimeInterval? = nil
+    
+    public typealias RingerModeDetectorCompletion = ((Bool) -> ())
+
+    private var listOfCompletions: [RingerModeDetectorCompletion] = []
+    
+    public func detect(_ e: @escaping RingerModeDetectorCompletion) {
+        guard soundId != 0 else {
+            e(false)
+            return
+        }
+        
+        self.listOfCompletions.append(e)
+        
+        if self.startTime == nil {
+            self.startTime = CACurrentMediaTime()
+            AudioServicesPlaySystemSound(self.soundId)
+        }
+    }
+    
+    private override init() {
+        super.init()
+        self.soundId = 1
+        
+        let result = AudioServicesCreateSystemSoundID(RingerModeDetector.soundUrl as CFURL, &soundId)
+        
+        if result == kAudioServicesNoError {
+            setupAudioServicesCompletion()
+        } else {
+            self.soundId = 0
+        }
+    }
+    
+    public func setupAudioServicesCompletion() -> Void {
+        let rowPointer = UnsafeMutableRawPointer(Unmanaged.passRetained(self).toOpaque())
```

Comment:
I have read the documentation for UnsafeMutableRawPointer and it mentioned that there is no automated memory management. Do we need to dispose rowPointer after completion?